### DISCUSSION TITLE
synology quickconnect + audiobookshelf

### DISCUSSION THREAD

**Discussion by lex1890** [advplyr/audiobookshelf#2480] [Q&A]
good day! tell me if it is possible to connect to audiobookshelf with quickconnect (by synology) ?
i dont have white ip address or ddns...

**Comment by treyg** [comment]
> good day! tell me if it is possible to connect to audiobookshelf with quickconnect (by synology) ?
> i dont have white ip address or ddns...

Directions for reverse proxy setup on Synology can be found here: https://github.com/advplyr/audiobookshelf/#synology-nas-reverse-proxy-setup-dsm-7quickconnect

**Reply by lex1890** [reply]
I found this instruction, but, for some unknown reason, it didn’t work for me (yes, my port is 13379)
![Снимок11](https://github.com/advplyr/audiobookshelf/assets/99123144/75de308f-9598-4d28-b8eb-8cd84c72f627)
![Снимок12](https://github.com/advplyr/audiobookshelf/assets/99123144/50d2144d-8f88-4920-8679-fee12c64b40e)
![Снимок44](https://github.com/advplyr/audiobookshelf/assets/99123144/0cd8f23e-05b2-48c9-949b-12a4440f3bbe)
![Снимок66](https://github.com/advplyr/audiobookshelf/assets/99123144/06e609f5-d57a-4650-954f-c677c957c0ec)

**Reply by nichwall** [reply]
Is your port open on your router?

**Reply by lex1890** [reply]
and what does router port forwarding have to do if we say that quick connect works even without forwarding

**Reply by treyg** [reply]
@lex1890 What happens why you try to go to `https://au.<yourid>.quickconnect.to`?

**Reply by lex1890** [reply]
![Снимок4](https://github.com/advplyr/audiobookshelf/assets/99123144/1041f95e-7674-4731-9096-2501565f2d24)

**Reply by treyg** [reply]
If you go to control panel -> security -> certificate, what do you see? You might need to add a new cert

**Comment by lex1890** [comment]
> If you go to control panel -> security -> certificate, what do you see? You might need to add a new cert
![Снимок5](https://github.com/advplyr/audiobookshelf/assets/99123144/dd34a450-0ab7-4f36-a69f-87aac796e941)

**Reply by treyg** [reply]
I think the issue is that you need to set up DDNS and use that for your reverse proxy. Steps 1-6 [here](https://kb.synology.com/en-us/DSM/help/DSM/AdminCenter/connection_ddns?version=7). Then when setting up your RP, use for example: `audiobookshelf.lex1890.synology.me` (or whatever you create) as the Hostname

**Reply by lex1890** [reply]
For ddns need have white ip adress, i have only grey ip. Quick connect works with grey ip.

**Reply by treyg** [reply]
You can't use a reverse proxy with quickconnect then. Your best bet is to use something like tailscale for remote access

**Comment by amwmedia** [comment]
Could the issue here be that you are trying to connect an SSL URL to a service that does not support it? If you change your quickconnect port to 80, does it work? Based on what I understand of quickconnect, it will allow you to make a connection to your NAS via a public address (bypassing the need for a static IP), but I don't think it handles SSL certs. So your browser is connecting to a SSL URL and expecting the server it reaches to provide a secure connection, but instead, it gets a basic HTTP response.

**Reply by lex1890** [reply]
I just followed the instructions and didn’t expect that those who wrote it knew what they were talking about. I'm not sure if quickconnect can be used this way

**Reply by amwmedia** [reply]
I used cloudflare tunnels for my ABS setup and it's great. Just throwing it out there 😉. https://www.youtube.com/watch?v=ZvIdFs3M5ic

**Reply by nichwall** [reply]
Streaming non-html content through a CloudFlare tunnel without hosting the data with CloudFlare is against CloudFlare's ToS. It works right now because they're not enforcing it, but at some point they will likely begin blocking streaming content (like Google Drive Unlimited enforcement).

CloudFlare Tunnels also only allow for 100 MB uploads if that affects you.

**Reply by amwmedia** [reply]
Can you cite where you see these restrictions in the ToS? Doesn't seem to jive with the other docs I've seen. They even list SSH and SMB as valid [use cases for tunnels](https://developers.cloudflare.com/cloudflare-one/connections/connect-networks/use-cases/).

**Reply by nichwall** [reply]
It's been discussed in Discord a number of times, but the main points are summarized in this blog post. I misspoke when I said "non HTML."

https://blog.cloudflare.com/updated-tos/

**Reply by amwmedia** [reply]
I'm failing to see the issue with tunnels. Most of the restrictions seem to be around the hosting of content within their network. Their tunnels product is specifically built for this sort of use-case (securely exposing internal services publically).

**Comment by lex1890** [comment]
Does anyone have any other ideas? Or is my idea not feasible in principle?
